### PR TITLE
Add TypeScript definition + transit/database support

### PR DESCRIPTION
**Pull request by brandonros**
It'd be a huge help to me and the organization I work for if we could get these changes/features merged and released.

It adds the following features:

Transit secret engine: https://www.vaultproject.io/api/secret/transit/index.html

Database secret engine: https://www.vaultproject.io/docs/secrets/databases/index.html

cc @eheikes @ChadKillingsworth

### CODE REVIEW

**Comment by brandonros** [comment]
@eheikes @chadk

**Comment by normano-cloud** [comment]
@kr1sp1n Can you pls look at this?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
+// Type definitions for node-vault 0.10
```
Comment (kr1sp1n) [inline index.d.ts]: There is no 0.10 version at the moment. Please remove version from comment.

Comment (brandonros) [reply index.d.ts]: Resolved. @kr1sp1n Thanks for taking a look into this!

**Comment by brandonros** [comment]
@kr1sp1n are we good here now?

**Comment by neilbryson** [comment]
Any updates on this ?

Also, the typings in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node-vault/index.d.ts) must be deprecated should this PR be merged.

**Comment by kr1sp1n** [comment]
@brandonros @neilbryson @normano-cloud Sorry for the late reply.
It's published now as v0.9.12 https://www.npmjs.com/package/node-vault 🎉

**Pull request merged**